### ISSUE TITLE
Hermes refresh for clients with small trusting period doesn't work

### ISSUE THREAD

**Issue by ancazamfir** [informalsystems/ibc-rs#1563]
## Crate
relayer

## Summary of Bug
Clients with small trusting period are not refreshed properly resulting in expiration. 
Maybe regression introduced by #1378

## Version
master

## Steps to Reproduce
Configure
```
trusting_period = '120s'
```
Create a new channel. Start hermes.
Refresh attempt comes late:
```
2021-11-11T08:27:30.196689Z  WARN ThreadId(192) failed to refresh client 'ibc-0 -> ibc-1:07-tendermint-3':
   0: client 07-tendermint-3 on chain id ibc-1 is expired or frozen
```

Works with
```
trusting_period = '180s'
```

## Acceptance Criteria
Either refresh these clients or don't allow trusting periods that are too small, i.e. same order of magnitude as the `60` seconds here:
https://github.com/informalsystems/ibc-rs/blob/52e0314050b5e984b740968fd90daaad83b5358e/relayer/src/worker/client.rs#L63

____

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate milestone (priority) applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by soareschen** [comment]
I think this is already fixed in #1664, with the test being able to run with 15s of trusting period. The key is that #1664 makes the refresh client task runs in parallel with other tasks such as detect misbehavior, so that it cannot be blocked by them. 

Note that the trusting period still cannot reliably go below 15s, because the chain handle is handling requests in serial. So even if the refresh client task can send a refresh transaction request on time, it can still get blocked in the chain handle's queue while the chain handle is busy handling other requests.

Also note that with the current design of the supervisor, the refresh client task do not get spawned unless there is at least one active channel. But the connection and channel handshake themselves can take much longer than 15s. The workaround in the integration test is to explicitly spawn the client refresh task itself before doing the channel handshake, and then shutdown the explicit tasks and spawn the supervisor for it to spawn the refresh task only after the channel handshake has been completed.

**Comment by adizere** [comment]
Closing as this is already fixed.
